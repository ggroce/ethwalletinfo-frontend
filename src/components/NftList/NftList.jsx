import './NftList.css';
import { useSelector } from 'react-redux';
import { getNfts } from '../../redux/wallet/wallet.selectors';
import NftCard from '../NftCard/NftCard';

const NftList = () => {
  const nfts = useSelector(getNfts);

  const renderNfts = () => {
    if (!nfts) {
      return <h3>Connect wallet to view NFTs</h3>;
    } else if (nfts.length === 0) {
      return <h3>No NFTs found</h3>;
    }
    return nfts.map(({ id, ...otherNftProps }) => (
      <NftCard key={id} {...otherNftProps} />
    ));
  };

  return (
    <>
      <div className="nftlist__scroll-wrapper">{renderNfts()}</div>
    </>
  );
};
export default NftList;
