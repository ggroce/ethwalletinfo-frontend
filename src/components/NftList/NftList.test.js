import { screen } from '@testing-library/react';

import NftList from './NftList';

import {
  renderWithRedux,
  initialStateWalletStore,
  editWalletStore,
} from '../../test/testing.utils';

test('<NftList />: takes a snapshot', () => {
  const { asFragment } = renderWithRedux(<NftList />, {
    initialState: initialStateWalletStore,
  });
  expect(
    asFragment(<NftList />, {
      initialState: initialStateWalletStore,
    })
  ).toMatchSnapshot();
});

test('<NftList />: no wallet connected', async () => {
  renderWithRedux(<NftList />, { initialState: initialStateWalletStore });
  expect(screen.queryByText('Connect wallet to view NFTs')).toBeInTheDocument();
});

test('<NftList />: tests list with no NFTs', async () => {
  const walletStoreZeroNft = editWalletStore({ nfts: [] });

  renderWithRedux(<NftList />, { initialState: walletStoreZeroNft });
  expect(screen.queryByText('No NFTs found')).toBeInTheDocument();
});

test('<NftList />: tests list with one NFT', async () => {
  const walletStoreOneNft = editWalletStore({
    nfts: [
      {
        id: '1',
        meta: {
          name: 'Random NFT',
          description: "This NFT is the bee's knees",
          image: {
            url: {
              PREVIEW: 'https://example.com/preview.png',
              ORIGINAL: 'https://example.com/original.png',
            },
          },
        },
      },
    ],
  });

  renderWithRedux(<NftList />, { initialState: walletStoreOneNft });
  expect(screen.queryByText('No NFTs found')).not.toBeInTheDocument();
  expect(
    screen.queryByText('Connect wallet to view NFTs')
  ).not.toBeInTheDocument();
  expect(screen.queryByText('Random NFT')).toBeInTheDocument();
});
