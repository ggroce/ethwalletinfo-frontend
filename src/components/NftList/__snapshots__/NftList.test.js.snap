// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NftList />: takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="nftlist__scroll-wrapper"
  >
    <h3>
      Connect wallet to view NFTs
    </h3>
  </div>
</DocumentFragment>
`;
